app/models/Menu.java

```diff
@@ -0,0 +1,104 @@
+package models;
+
+import com.avaje.ebean.Model;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonView;
+import org.apache.commons.io.FilenameUtils;
+import play.mvc.Http;
+import utilities.AWSManager;
+import utilities.PersistenceManager;
+import utilities.Validation;
+import utilities.View;
+
+import javax.persistence.*;
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.UUID;
+
+/**
+ * Created by Ejub on 2.4.2016.
+ */
+@Entity
+@Table(name = "abh_menu")
+public class Menu extends Model implements Validation{
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @JsonView(View.BasicDetails.class)
+    private long menuId;
+    @JsonIgnore
+    @ManyToOne()
+    private Restaurant restaurant;
+    @Transient
+    @JsonIgnore
+    private File file;
+    private String name;
+
+    public static Model.Finder<String, Menu> find = new Model.Finder<String, Menu>(String.class, Menu.class);
+
+    public Menu(Http.Request request, Restaurant restaurant) {
+        this.restaurant = restaurant;
+        file = request.body().asMultipartFormData().getFile("upload").getFile();
+        String fileExtension = FilenameUtils.getExtension(request.body().asMultipartFormData().getFile("upload").getFilename());
+        name = UUID.randomUUID().toString() + "." + fileExtension;
```

Comment:
wrapper class should handle this and extension should be red differently (as for photo).... also, please add validations for menu upload to PDF only and JPG/PNG for photos